File: LeetCode/move-zeroes.cpp
// https://leetcode.com/problems/move-zeroes/

class Solution
{
public:
	void swapNumbers(int &x, int &y)
	{
		int temp = x;
		x = y;
		y = temp;
	}

	void moveZeroes(vector<int> &nums)
	{
		int numsLength = nums.size();
		int zeroPointer = 0;

		for (int i = 0; i < numsLength; i++)
		{
			if (nums[i] != 0)
			{
				swapNumbers(nums[i], nums[zeroPointer]);
				zeroPointer++;
			}
		}

		for (auto itr = nums.begin(); itr != nums.end(); itr++)
		{
			cout << *itr << " ";
		}
	}
};

File: LeetCode/Java/movezeroes/MoveZeroes.java
package movezeroes;

import java.util.Arrays;

public class MoveZeroes {
	public static void main(String[] args) {
		Solution sol = new Solution();
		sol.moveZeroes(new int[]{0, 1, 0, 3, 12});
	}
}

// https://leetcode.com/problems/move-zeroes/

//IMPORTANT!! Submit Code Region Begin(Do not remove this line)
class Solution {
	public void moveZeroes(int[] nums) {
		int insertPos = 0;

		for (int i = 0; i < nums.length; i++) {
			if (nums[i] != 0) {
				if (i != insertPos) {
					nums[insertPos] = nums[i];
					nums[i] = 0;
				}
				insertPos++;
			}
		}

		System.out.println(Arrays.toString(nums));
	}
}

//IMPORTANT!! Submit Code Region End(Do not remove this line)